google_compute_engine/network_utils.py

```diff
@@ -38,11 +44,16 @@ def _CreateInterfaceMap(self):
       dict, string MAC addresses mapped to the string network interface name.
     """
     interfaces = {}
-    for interface in os.listdir('/sys/class/net'):
+    for interface in netifaces.interfaces():
       try:
-        mac_address = open(
-            '/sys/class/net/%s/address' % interface).read().strip()
-      except (IOError, OSError) as e:
+        mac_address = (
+            netifaces.ifaddresses(interface)[netifaces.AF_LINK][0]['addr'])
```

Comment:
right, I thought we didn't have to call it in a safe way, as it would though and exception that would be caught. If in the end mac_address is None, the code will raise an exception anyway. But I don't mind changing if you prefer like this.